### PR TITLE
Adds Report formspec

### PR DESCRIPTION
**Pull request by Anthony-De**
Adds a formspec for reports with builtin messages.

### CODE REVIEW

**Comment by appgurueu** [comment]
That's called a formspec, not a HUD.

File: mods/ctf/ctf_reports/init.lua [hunk 1]
```diff
@@ -111,29 +111,47 @@ function ctf_reports.send_report(report)
...
-		if param == "" then
-			return false, "Please add a message to your report. " ..
-				"If it's about (a) particular player(s), please also include their name(s)."
+		local connected_players = {}
+		for _,player in pairs(minetest.get_connected_players()) do
+			if player:is_player() and player:get_player_name() then
+				table.insert(connected_players, player:get_player_name())
+			end
 		end
+		local report_formspec = ""..
```
Comment (appgurueu) [inline mods/ctf/ctf_reports/init.lua:132]: `""..` is unneeded, set formspec version properly so you don't have to use weird offsets in the following lines.

Comment (Anthony-De) [reply mods/ctf/ctf_reports/init.lua:132]: > You should consider escaping some text parts
Which means?

File: mods/ctf/ctf_reports/init.lua [hunk 2]
```diff
@@ -111,29 +111,47 @@ function ctf_reports.send_report(report)
...
+
 minetest.register_chatcommand("report", {
 	func = function(name, param)
-		param = param:trim()
-		if param == "" then
-			return false, "Please add a message to your report. " ..
-				"If it's about (a) particular player(s), please also include their name(s)."
+		local connected_players = {}
+		for _,player in pairs(minetest.get_connected_players()) do
+			if player:is_player() and player:get_player_name() then
```
Comment (appgurueu) [inline mods/ctf/ctf_reports/init.lua:128]: Both checks are unneeded. Do:
```lua
for index, player in ipairs(minetest.get_connected_players()) do connected_players[index] = player:get_player_name() end
```

File: mods/ctf/ctf_reports/init.lua [hunk 3]
```diff
@@ -111,29 +111,47 @@ function ctf_reports.send_report(report)
 	end
 end
 
+local report_reasons = {
+	"chat abuse",
```
Comment (appgurueu) [inline mods/ctf/ctf_reports/init.lua:115]: change to "chat abusing" or "abusing chat" to make it fit in

File: mods/ctf/ctf_reports/init.lua (hunk 1 above)
Comment (Anthony-De) [reply mods/ctf/ctf_reports/init.lua:132]: > set formspec version properly 

How would I do this?

**Comment by LoneWolfHT** [comment]
```cd capturetheflag/mods/ctf/ctf_map/ctf_map_core/maps/
git checkout master && git remote update origin && git checkout origin/master
```

File: mods/ctf/ctf_reports/init.lua [hunk 4]
```diff
@@ -118,7 +118,8 @@ local report_reasons = {
 	"spawn killing",
 	"hacking",
 	"cross teaming",
-	"base griefing"
+	"base griefing",
+	"inappropriate"
```
Comment (appgurueu) [inline mods/ctf/ctf_reports/init.lua:122]: inappropriate what?

Comment (Anthony-De) [reply mods/ctf/ctf_reports/init.lua:122]: Name, building, behavior

File: mods/ctf/ctf_reports/init.lua [hunk 5]
```diff
@@ -111,29 +111,46 @@ function ctf_reports.send_report(report)
...
 
+local report_reasons = {
+	"abusing chat",
+	"swearing",
+	"asking for personal info",
+	"spawn killing",
+	"hacking",
+	"cross teaming",
+	"base griefing",
+	"inappropriate"
```
Comment (Anthony-De) [inline mods/ctf/ctf_reports/init.lua:122]:
```suggestion
	"being inappropriate"
```

**Comment by DiamondPlane** [comment]
Suggestion: Edit it to something like this:
``ctf_reports.send_report( minetest.colorize(tcolor.css, player:get_player_name() .. " reported: " ..fields.username_dropdown .." was "..fields.reason_dropdown)`` (Requires tcolor)

**Comment by DiamondPlane** [comment]
Suggestion: Make it "<player> reported: <player> was <report>" because it doesn't say that someone reported.

**Comment by xenonca** [comment]
TBH I prefer quick reports that don't require me to click and type stuff into a formspec. Especially in-game a "chat based" report can be done much faster IMO

**Comment by appgurueu** [comment]
> TBH I prefer quick reports that don't require me to click and type stuff into a formspec. Especially in-game a "chat based" report can be done much faster IMO

Can be done easily: If no argument is provided to `/report`, it will open said formspec.

File: mods/ctf/ctf_reports/init.lua [hunk 6]
```diff
@@ -111,29 +111,62 @@ function ctf_reports.send_report(report)
...
 
+local report_reasons = {
+	"abusing chat",
+	"swearing",
+	"asking for personal info",
+	"spawn killing",
+	"hacking",
+	"cross teaming",
+	"base griefing",
+	"being inappropriate"
```
Comment (Panquesito7) [inline mods/ctf/ctf_reports/init.lua:122]: Maybe you should provide another customizable option (i.e.: `Other (please describe`).

**Review by Panquesito7 — COMMENTED** [review]
Why is the submodule `shooter` being updated in this PR, by the way? 🤔

**Pull request closed without merging**